projects/dsp-ui/src/lib/viewer/representation/upload-form/upload-form.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { NotificationService } from '../../../action/services/notification.service';
+import { UploadedFileResponse, UploadFileService } from '../../services/upload-file.service';
+
+@Component({
+    selector: 'dsp-upload-form',
+    templateUrl: './upload-form.component.html',
+    styleUrls: ['./upload-form.component.scss']
+})
+export class UploadFormComponent implements OnInit {
+
+    @Input() readonly resourceTyoe = 'Image'; // only StillImageRepresentation supported so far
+    @Output() file: File;
+
+    readonly fromLabels = {
+        upload: 'Upload file',
+        drag_drop: 'Drag and drop or click to upload'
+    };
+    form: FormGroup;
+    get fileControl() { return this.form.get('file') as FormControl; }
+    isLoading = false;
+    thumbnaillUrl: string;
+
+    constructor(
+        private readonly _fb: FormBuilder,
+        private readonly _ns: NotificationService,
+        private readonly _ufs: UploadFileService
+    ) { }
+
+    ngOnInit(): void {
+        this.initializeForm();
+    }
+
+    initializeForm(): void {
```

Comment:
Could you please add some documentation to the component's methods?

Reply:
Do you mean comments over some lines of methods?

Reply:
the js doc style, for example:

```
    /**
     * Determines if the given value can be edited.
     *
     * @param valueTypeOrClass the type or class of the given value.
     * @param value the given value.
     */
    isReadOnly(valueTypeOrClass: string, value: ReadValue): boolean {
    ...
     }
```

in most IDEs there is some support for this with the params etc.

Reply:
I think there is no need to annotate the return type or whatever is already there in TypeScript.

Reply:
True, there this plugin is also available in vscode. I can install and use it, however I'm rather fan of descriptive code methodology than comments inside methods.

Reply:
Fixed with 679fc9b.

Reply:
Thanks! Could you add a short explanation for the method params too (like in the example given [above](https://github.com/dasch-swiss/dsp-ui-lib/pull/218#discussion_r517890904))?

Reply:
Done with 895d3b7.